### PR TITLE
Deploy

### LINKED ISSUE #40: Deploy project
Backend: google app engine?
Frontend: netlify?

### PR DESCRIPTION
**Pull request by carletex**
**Backend**: We are using Google App Engine (nodejs14 env) to deploy the backend. I made the necessary changes on the backend code (env vars, firabase instanciation) + added the App engine config file: `app.yaml`.

@sogasg @dgrcode you can read in the updated README the new way to configure Firabase for your local machine (basically setting the `GOOGLE_APPLICATION_CREDENTIALS` env var)

I deployed it manually using the google cloud console, but we could easily integrate it with Google Cloud Build to have CD. For that we'll need to give OAuth permission to the Cloud Build APP (need a repo owner for that.)

**Frontend**: We are using Netlify to deploy the frontend. I added a env var to define the backend_url and added a `_redirects` file to make React Router work with Netlify. To deploy to netlify we'll need to connect this repo with it (need a repo owner for that). Already requested:

image

For testing purposes, I've forked the repo and was be able to deploy to Netlify:

https://wonderful-kirch-4ab41a.netlify.app/

Fixes #40

### CODE REVIEW

**Comment by carletex** [comment]
Hey @over9001 Did you get a Netlify request? I need it to be able to deploy this repo's frontend to Netlify.

image

File: README.md [hunk 1]
```diff
@@ -46,9 +46,9 @@ At this point, you should have the app available at <http://localhost:3000>. By
 
 ## Firebase Setup (optional)
 
-If you want to use Firebase (firestore) for data storage, you'll need to create a Firebase project and place the service account key configuration in `packages/backend/firebaseServiceAccountKey.json`. You can generate and download the file at <https://console.cloud.google.com/>, under IAM & Admin > Service Accounts > Keys.
+If you want to use Firebase (firestore) for data storage, you'll need to create a Firebase project and download the service account key configuration. You can generate and download the file at <https://console.cloud.google.com/>, under IAM & Admin > Service Accounts > Keys.
```
Comment (sogasg) [inline README.md:49]:
```suggestion
If you want to use Firebase (firestore) for data storage, you'll need to create a Firebase project and download the service account key configuration. You can generate and download the service account file at <https://console.cloud.google.com/> by 1.) select your Firebase project, 2.) go to IAM & Admin > Service Accounts, 3.) create a service account or click one that is already created, 4.) go to keys of that account, and 5.) Add Key > Create key and select a JSON key type.
```

File: README.md [hunk 2]
```diff
@@ -46,9 +46,9 @@ At this point, you should have the app available at <http://localhost:3000>. By
...
 
-Then you will have to create a `.env` file in `packages/backend/` and add `DATABASE_SERVICE=firebase` (see a sample here `packages/backend/.env.sample`).
+Then you will have to create a `.env` file in `packages/backend/` with `DATABASE_SERVICE=firebase` the path to the service account on `GOOGLE_APPLICATION_CREDENTIALS` (see a sample here `packages/backend/.env.sample`).
```
Comment (sogasg) [inline README.md:51]:
```suggestion
Then you will have to create a `.env` file in `packages/backend/` with `DATABASE_SERVICE=firebase` and add the full path to the service account file in `GOOGLE_APPLICATION_CREDENTIALS` (see a sample here `packages/backend/.env.sample`).
```

**Review by sogasg — CHANGES_REQUESTED** [review]
Nice work :) 

I got it running both locally and with the Firebase DB. However, I did not manage to do the Firebase setup solely based on the information in the readme. I suggested some updates.

Also, I did not test the deployment to the Google app engine only ran it locally.

**Comment by carletex** [comment]
Thanks @sogasg !! Accepted your changes, I think it's clear now.

> Also, I did not test the deployment to the Google app engine only ran it locally.

Yep, hard to test unless you create a project yourself. This front-end https://wonderful-kirch-4ab41a.netlify.app/ is making requests to the GApp Engine backend.

**Review by sogasg — APPROVED** [review]
Nice 😄

**Pull request merged**